test/common/network/socket_option_test_harness.h

```diff
@@ -0,0 +1,67 @@
+#include "common/network/address_impl.h"
+#include "common/network/socket_option_impl.h"
+
+#include "test/mocks/api/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/logging.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+
+#include "gtest/gtest.h"
+
+using testing::Invoke;
+using testing::NiceMock;
+using testing::Return;
+using testing::_;
+
+namespace Envoy {
+namespace Network {
+namespace {
+
+class SocketOptionTestHarness : public testing::Test {
```

Comment:
Yep, this looks good. As a nit, I would probably call it `SocketOptionTestBase` or `SocketOptionTest`.